### PR TITLE
fix decision forest incorrect number of features in prediction data bug

### PR DESCRIPTION
**Pull request by PivovarA**
Decision forest prediction worked with any number of features.
Added a match check for number of features in the training data and in the data for prediction

### CODE REVIEW

**Comment by PivovarA** [comment]
make test

**Comment by PivovarA** [comment]
make test

**Comment by PivovarA** [comment]
make test

**Comment by PivovarA** [comment]
make test

**Comment by PivovarA** [comment]
https://ecmd.inn.intel.com/commander/link/jobDetails/jobs/ea41c26b-e3e6-f198-b1d0-8cdcd4b723a1?linkPageType=jobDetails&objectLocator=%2Fjobs%2Fea41c26b-e3e6-f198-b1d0-8cdcd4b723a1&setPageSize=1&pageSize=500&firstElement=1&firstPage=1

File: algorithms/kernel/dtrees/forest/classification/df_classification_train_container.h [hunk 1]
```diff
@@ -66,6 +66,8 @@ services::Status BatchContainer<algorithmFPType, method, cpu>::compute()
 
     decision_forest::classification::Model * m = result->get(classifier::training::model).get();
 
+    m->setNFeatures(x->getNumberOfColumns());
```
Comment (ShvetsKS) [inline algorithms/kernel/dtrees/forest/classification/df_classification_train_container.h]: Seems the same problem is for DF regression. Could you create issue on it or fix it also in this PR?

Comment (PivovarA) [reply algorithms/kernel/dtrees/forest/classification/df_classification_train_container.h]: This issue has already been created: https://github.com/intel/daal/issues/307.
I will create a new pr with a fix for DF regression.

File: include/algorithms/decision_forest/decision_forest_classification_model_builder.h [hunk 2]
```diff
@@ -129,6 +129,17 @@ class DAAL_EXPORT ModelBuilder
...
 
+
+    void setNFeatures(size_t nFeatures)
+    {
+        if(!_model.get())
+        {
+            _status |= services::ErrorNullModel;
+            services::throwIfPossible(_status);
+        }
+        _model->setNFeatures(nFeatures);
```
[the remark this answers is not in the document]
Comment (PivovarA) [reply include/algorithms/decision_forest/decision_forest_classification_model_builder.h]: done

**Comment by PivovarA** [comment]
BC job https://ecmd.inn.intel.com/commander/link/jobDetails/jobs/ea434d65-e942-f1fc-89c3-8cdcd4b723a1

**Pull request merged**